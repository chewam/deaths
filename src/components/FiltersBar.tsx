import { useRouter } from "next/router"

import useFilters from "@/services/filters"
import Genders from "@/components/filters/Genders"
import AgeGroups from "@/components/filters/AgeGroups"

const Filters = (): JSX.Element => {
  const { route } = useRouter()
  const [filters, setFilters] = useFilters()

  const { ageGroup, gender } = filters as Filters

  const handleGenderChange = (gender: Gender) =>
    setFilters({ ageGroup, gender })

  const handleAgeGroupChange = (ageGroup: [number, number]) =>
    setFilters({ ageGroup, gender })

  return (
    <div className="container mx-auto mt-24 px-6">
      {route !== "/" && (
        <div className="filters">
          <AgeGroups onChange={handleAgeGroupChange} />
          <Genders onChange={handleGenderChange} />
        </div>
      )}
    </div>
  )
}

export default Filters
