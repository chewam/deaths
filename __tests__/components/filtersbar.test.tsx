import messages from "@/lang/fr.json"
import FiltersBar from "@/components/FiltersBar"
import { IntlProvider } from "react-intl"
import { ThemeProvider } from "next-themes"
import { render } from "@testing-library/react"

Object.defineProperty(window, "matchMedia", {
  writable: true,
  value: jest.fn().mockImplementation((query) => ({
    matches: false,
    media: query,
    onchange: null,
    addListener: jest.fn(), // Deprecated
    removeListener: jest.fn(), // Deprecated
    addEventListener: jest.fn(),
    removeEventListener: jest.fn(),
    dispatchEvent: jest.fn(),
  })),
})

jest.mock("next/router", () => ({
  useRouter() {
    return {
      route: "/",
      locale: "fr",
      defaultLocale: "en",
    }
  },
}))

/* eslint @typescript-eslint/no-var-requires: "off" */
const useRouter = jest.spyOn(require("next/router"), "useRouter")

describe("Filters Bar", () => {
  test("should create a visible Filters Bar", () => {
    useRouter.mockImplementation(() => ({
      route: "/comparison",
    }))
    const { asFragment } = render(
      <IntlProvider locale="fr" messages={messages}>
        <ThemeProvider defaultTheme="light">
          <FiltersBar />
        </ThemeProvider>
      </IntlProvider>
    )
    expect(asFragment()).toMatchSnapshot()
  })

  test("should create a hidden Filters Bar", () => {
    useRouter.mockImplementation(() => ({
      route: "/",
    }))
    const { asFragment } = render(
      <IntlProvider locale="fr" messages={messages}>
        <ThemeProvider
          defaultTheme="dark"
          forcedTheme="dark"
          themes={["light", "dark"]}
        >
          <FiltersBar />
        </ThemeProvider>
      </IntlProvider>
    )
    expect(asFragment()).toMatchSnapshot()
  })
})
