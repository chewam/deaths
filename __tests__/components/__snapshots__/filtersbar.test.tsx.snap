// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filters Bar should create a hidden Filters Bar 1`] = `
<DocumentFragment>
  <div
    class="filters"
  >
    <div
      class="rc-slider rc-slider-horizontal rc-slider-with-marks"
    >
      <div
        class="rc-slider-rail"
      />
      <div
        class="rc-slider-track rc-slider-track-1"
        style="left: 0%; width: 100%;"
      />
      <div
        class="rc-slider-step"
      >
        <span
          class="rc-slider-dot rc-slider-dot-active"
          style="left: 0%; transform: translateX(-50%);"
        />
        <span
          class="rc-slider-dot rc-slider-dot-active"
          style="left: 9.090909090909092%; transform: translateX(-50%);"
        />
        <span
          class="rc-slider-dot rc-slider-dot-active"
          style="left: 18.181818181818183%; transform: translateX(-50%);"
        />
        <span
          class="rc-slider-dot rc-slider-dot-active"
          style="left: 27.27272727272727%; transform: translateX(-50%);"
        />
        <span
          class="rc-slider-dot rc-slider-dot-active"
          style="left: 36.36363636363637%; transform: translateX(-50%);"
        />
        <span
          class="rc-slider-dot rc-slider-dot-active"
          style="left: 45.45454545454545%; transform: translateX(-50%);"
        />
        <span
          class="rc-slider-dot rc-slider-dot-active"
          style="left: 54.54545454545454%; transform: translateX(-50%);"
        />
        <span
          class="rc-slider-dot rc-slider-dot-active"
          style="left: 63.63636363636363%; transform: translateX(-50%);"
        />
        <span
          class="rc-slider-dot rc-slider-dot-active"
          style="left: 72.72727272727273%; transform: translateX(-50%);"
        />
        <span
          class="rc-slider-dot rc-slider-dot-active"
          style="left: 81.81818181818183%; transform: translateX(-50%);"
        />
        <span
          class="rc-slider-dot rc-slider-dot-active"
          style="left: 90.9090909090909%; transform: translateX(-50%);"
        />
        <span
          class="rc-slider-dot rc-slider-dot-active"
          style="left: 100%; transform: translateX(-50%);"
        />
      </div>
      <div
        aria-disabled="false"
        aria-valuemax="110"
        aria-valuemin="0"
        aria-valuenow="0"
        class="rc-slider-handle rc-slider-handle-1"
        role="slider"
        style="left: 0%; transform: translateX(-50%);"
        tabindex="0"
      />
      <div
        aria-disabled="false"
        aria-valuemax="110"
        aria-valuemin="0"
        aria-valuenow="110"
        class="rc-slider-handle rc-slider-handle-2"
        role="slider"
        style="left: 100%; transform: translateX(-50%);"
        tabindex="0"
      />
      <div
        class="rc-slider-mark"
      >
        <span
          class="rc-slider-mark-text rc-slider-mark-text-active"
          style="left: 0%; transform: translateX(-50%);"
        >
          0
        </span>
        <span
          class="rc-slider-mark-text rc-slider-mark-text-active"
          style="left: 9.090909090909092%; transform: translateX(-50%);"
        >
          10
        </span>
        <span
          class="rc-slider-mark-text rc-slider-mark-text-active"
          style="left: 18.181818181818183%; transform: translateX(-50%);"
        >
          20
        </span>
        <span
          class="rc-slider-mark-text rc-slider-mark-text-active"
          style="left: 27.27272727272727%; transform: translateX(-50%);"
        >
          30
        </span>
        <span
          class="rc-slider-mark-text rc-slider-mark-text-active"
          style="left: 36.36363636363637%; transform: translateX(-50%);"
        >
          40
        </span>
        <span
          class="rc-slider-mark-text rc-slider-mark-text-active"
          style="left: 45.45454545454545%; transform: translateX(-50%);"
        >
          50
        </span>
        <span
          class="rc-slider-mark-text rc-slider-mark-text-active"
          style="left: 54.54545454545454%; transform: translateX(-50%);"
        >
          60
        </span>
        <span
          class="rc-slider-mark-text rc-slider-mark-text-active"
          style="left: 63.63636363636363%; transform: translateX(-50%);"
        >
          70
        </span>
        <span
          class="rc-slider-mark-text rc-slider-mark-text-active"
          style="left: 72.72727272727273%; transform: translateX(-50%);"
        >
          80
        </span>
        <span
          class="rc-slider-mark-text rc-slider-mark-text-active"
          style="left: 81.81818181818183%; transform: translateX(-50%);"
        >
          90
        </span>
        <span
          class="rc-slider-mark-text rc-slider-mark-text-active"
          style="left: 90.9090909090909%; transform: translateX(-50%);"
        >
          100
        </span>
        <span
          class="rc-slider-mark-text rc-slider-mark-text-active"
          style="left: 100%; transform: translateX(-50%);"
        >
          110+
        </span>
      </div>
    </div>
    <div
      class="genders"
    >
      <i
        aria-checked="false"
        aria-hidden="true"
        class="ri-men-line "
        role="checkbox"
        tabindex="0"
        title="hommes"
      />
      <i
        aria-checked="false"
        aria-hidden="true"
        class="ri-women-line "
        role="checkbox"
        tabindex="0"
        title="femmes"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`Filters Bar should create a visible Filters Bar 1`] = `
<DocumentFragment>
  <div
    class="filters"
  >
    <div
      class="rc-slider rc-slider-horizontal rc-slider-with-marks"
    >
      <div
        class="rc-slider-rail"
      />
      <div
        class="rc-slider-track rc-slider-track-1"
        style="left: 0%; width: 100%;"
      />
      <div
        class="rc-slider-step"
      >
        <span
          class="rc-slider-dot rc-slider-dot-active"
          style="left: 0%; transform: translateX(-50%);"
        />
        <span
          class="rc-slider-dot rc-slider-dot-active"
          style="left: 9.090909090909092%; transform: translateX(-50%);"
        />
        <span
          class="rc-slider-dot rc-slider-dot-active"
          style="left: 18.181818181818183%; transform: translateX(-50%);"
        />
        <span
          class="rc-slider-dot rc-slider-dot-active"
          style="left: 27.27272727272727%; transform: translateX(-50%);"
        />
        <span
          class="rc-slider-dot rc-slider-dot-active"
          style="left: 36.36363636363637%; transform: translateX(-50%);"
        />
        <span
          class="rc-slider-dot rc-slider-dot-active"
          style="left: 45.45454545454545%; transform: translateX(-50%);"
        />
        <span
          class="rc-slider-dot rc-slider-dot-active"
          style="left: 54.54545454545454%; transform: translateX(-50%);"
        />
        <span
          class="rc-slider-dot rc-slider-dot-active"
          style="left: 63.63636363636363%; transform: translateX(-50%);"
        />
        <span
          class="rc-slider-dot rc-slider-dot-active"
          style="left: 72.72727272727273%; transform: translateX(-50%);"
        />
        <span
          class="rc-slider-dot rc-slider-dot-active"
          style="left: 81.81818181818183%; transform: translateX(-50%);"
        />
        <span
          class="rc-slider-dot rc-slider-dot-active"
          style="left: 90.9090909090909%; transform: translateX(-50%);"
        />
        <span
          class="rc-slider-dot rc-slider-dot-active"
          style="left: 100%; transform: translateX(-50%);"
        />
      </div>
      <div
        aria-disabled="false"
        aria-valuemax="110"
        aria-valuemin="0"
        aria-valuenow="0"
        class="rc-slider-handle rc-slider-handle-1"
        role="slider"
        style="left: 0%; transform: translateX(-50%);"
        tabindex="0"
      />
      <div
        aria-disabled="false"
        aria-valuemax="110"
        aria-valuemin="0"
        aria-valuenow="110"
        class="rc-slider-handle rc-slider-handle-2"
        role="slider"
        style="left: 100%; transform: translateX(-50%);"
        tabindex="0"
      />
      <div
        class="rc-slider-mark"
      >
        <span
          class="rc-slider-mark-text rc-slider-mark-text-active"
          style="left: 0%; transform: translateX(-50%);"
        >
          0
        </span>
        <span
          class="rc-slider-mark-text rc-slider-mark-text-active"
          style="left: 9.090909090909092%; transform: translateX(-50%);"
        >
          10
        </span>
        <span
          class="rc-slider-mark-text rc-slider-mark-text-active"
          style="left: 18.181818181818183%; transform: translateX(-50%);"
        >
          20
        </span>
        <span
          class="rc-slider-mark-text rc-slider-mark-text-active"
          style="left: 27.27272727272727%; transform: translateX(-50%);"
        >
          30
        </span>
        <span
          class="rc-slider-mark-text rc-slider-mark-text-active"
          style="left: 36.36363636363637%; transform: translateX(-50%);"
        >
          40
        </span>
        <span
          class="rc-slider-mark-text rc-slider-mark-text-active"
          style="left: 45.45454545454545%; transform: translateX(-50%);"
        >
          50
        </span>
        <span
          class="rc-slider-mark-text rc-slider-mark-text-active"
          style="left: 54.54545454545454%; transform: translateX(-50%);"
        >
          60
        </span>
        <span
          class="rc-slider-mark-text rc-slider-mark-text-active"
          style="left: 63.63636363636363%; transform: translateX(-50%);"
        >
          70
        </span>
        <span
          class="rc-slider-mark-text rc-slider-mark-text-active"
          style="left: 72.72727272727273%; transform: translateX(-50%);"
        >
          80
        </span>
        <span
          class="rc-slider-mark-text rc-slider-mark-text-active"
          style="left: 81.81818181818183%; transform: translateX(-50%);"
        >
          90
        </span>
        <span
          class="rc-slider-mark-text rc-slider-mark-text-active"
          style="left: 90.9090909090909%; transform: translateX(-50%);"
        >
          100
        </span>
        <span
          class="rc-slider-mark-text rc-slider-mark-text-active"
          style="left: 100%; transform: translateX(-50%);"
        >
          110+
        </span>
      </div>
    </div>
    <div
      class="genders"
    >
      <i
        aria-checked="false"
        aria-hidden="true"
        class="ri-men-line "
        role="checkbox"
        tabindex="0"
        title="hommes"
      />
      <i
        aria-checked="false"
        aria-hidden="true"
        class="ri-women-line "
        role="checkbox"
        tabindex="0"
        title="femmes"
      />
    </div>
  </div>
</DocumentFragment>
`;
